src/main/java/uk/ac/ebi/ampt2d/accession/sha1/SHA1AccessionGenerator.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.sha1;
+
+import uk.ac.ebi.ampt2d.accession.HashableMessage;
+import uk.ac.ebi.ampt2d.accession.SingleAccessionGenerator;
+
+public class SHA1AccessionGenerator<T extends HashableMessage<String>> extends SingleAccessionGenerator<T, String> {
+
+    @Override
+    protected String generateAccession(T object) {
+        String accession = Sha1Util.generateSha1Accession(object.getHashableMessage().getBytes());
```

Comment:
The sha1 calculation is kind of a generic utility rather than accession related thats why i have put in a seperate utility class. May be i can rename the util function to generatesha1fromBytes ??

Reply:
But it is only used once in the whole code base, that is why I don't see the point of having it in a separate class.

Reply:
ok then let me then leave it for refactoring ticket